toxbat/requirements.py

```diff
@@ -62,13 +74,15 @@ def are_requirements_changed(config):
 
     def build_fpath_for_previous_version(fname):
         tox_dir = config.config.toxworkdir.strpath
-        fname = '{0}.{1}.previous'.format(fname.replace('/', '-'), config.envname)
+        envdirkey = _str_to_sha1hex(str(config.envdir))
+        fname = '{0}.{1}.previous'.format(fname.replace('/', '-'), envdirkey)
         return os.path.join(tox_dir, fname)
 
     requirement_files = map(parse_requirements_fname, deps)
     return any(
         is_changed(reqfile, build_fpath_for_previous_version(reqfile))
-        for reqfile in requirement_files if reqfile and os.path.isfile(reqfile))
+        for reqfile in requirement_files
```

Comment:
When calling `any()` with a generator, once a truthy value is encountered, any remaining items from the generator are not processed.

```
>>> def true():
...  print('true')
...  return True
...
>>> def false():
...  print('false')
...  return False
...
>>> any(fn() for fn in (true, false))
true
True
>>> any(fn() for fn in (false, true))
false
true
True
```

So when there is more than one requirements file, if the first requirements file has changed (or is missing), then any subsequent requirements hash files are not built. When using a single tox venv, this results in X venvs being created/recreated where X is the number of requirements file referenced in  tox `deps`.